core/src/test/scala/org/apache/spark/sql/txn/TxnTestSuite.scala

```diff
@@ -0,0 +1,234 @@
+package org.apache.spark.sql.txn
+
+import java.sql.{DriverManager, SQLException}
+import com.pingcap.tikv.kvproto.Kvrpcpb.IsolationLevel
+import org.apache.spark.sql.BaseTiSparkSuite
+import org.apache.spark.sql.catalyst.util.resourceToString
+
+// TODO: this test is not so useful at all
+// what I do is to construct a very long-running write operation
+// , a very long-running read operation and corresponding
+// two quick write and read operation run multiple times(20-100)
+// concurrently with multithreading. While the txns' prewrite
+// and commit finishes very quickly and almost synchronously.
+// What makes the resolveLock almost doesn't happen. What I think
+// the most useful way is to implement a mock kv, which delay the time
+// between prewrite and commit which cause the txn to rollback
+class TxnTestSuite extends BaseTiSparkSuite {
+  protected final val sumString = resourceToString(
+    s"resolveLock-test/sum_account.sql",
+    classLoader = Thread.currentThread().getContextClassLoader
+  )
+  protected final val q1String = resourceToString(
+    s"resolveLock-test/q1.sql",
+    classLoader = Thread.currentThread().getContextClassLoader
+  )
+  protected final val q2String = resourceToString(
+    s"resolveLock-test/q2.sql",
+    classLoader = Thread.currentThread().getContextClassLoader
+  )
+  protected final val giveString = resourceToString(
+    s"resolveLock-test/1_give.sql",
+    classLoader = Thread.currentThread().getContextClassLoader
+  )
+  protected final val getString = resourceToString(
+    s"resolveLock-test/2_get.sql",
+    classLoader = Thread.currentThread().getContextClassLoader
+  )
+  protected final val accountString = resourceToString(
+    s"resolveLock-test/1_account.sql",
+    classLoader = Thread.currentThread().getContextClassLoader
+  )
+  protected final val rnd = new scala.util.Random
+
+  /**
+   * query to tidb with jdbc and txn style
+   *
+   * @param query all querys run in the txn
+   * @param wait whether wait every 2 seconds
+   * @return Unit
+   */
+  protected def queryTIDBTxn(query: Seq[String], wait: Boolean): Unit = {
+    val conn = DriverManager.getConnection(jdbcUrl, "root", "")
```

Comment:
you might want to wrap this getConnection inside `SharedSQLContext` because users should be able to configure username and password via `properties`. For instance, calling `createJDBCConnection()` will create a new Connection using `DriverManager.getConnection(jdbcUrl, jdbcUsername, jdbcPassword)` with code from 
https://github.com/pingcap/tispark/blob/de938f0037863bb6e36611d20fa2c3dc8de5b7ab/core/src/test/scala/org/apache/spark/sql/test/SharedSQLContext.scala#L171-L173